### PR TITLE
chore: add dotnet example for presigned url

### PR DESCRIPTION
**Pull request by wandaitzuchen**
Tested against alpha.
```
% SIGNING_KEY="{\"p\":\"_GMb17SGMjnEUkkFLFPa5INFGEt8KIQBbkTvyR_LlYnGUgqURtajc-DzYFmhLEe35cRKxJ5qYa5NlumFCIocGsAZsPUSf5gZ2k685W9Gw4RVQ6_YT7ch2T8ahfN5rFDfw5RpaCyU1b8IGq7JSHr-3um3ILWv9LfDrrmLBcm6JGE\",\"kty\":\"RSA\",\"q\":\"2WdDgziADCKTiHqmr3bwOndtGqldd9CSIygPhLn1McTeP6rAx5G27wd_SV1J4EeGtfI_m6Az4LqFDJOXF2Dyae_HxEVMrnTf6nEk8sBFKm85DugBR-NM9BU7cSeZ2cq3VH9AuApAREekhBWPQAyK0MIZz1Ot0dnvTH-RleBtwqU\",\"d\":\"DTbYPuviNJw_kWh2aaXg4BtQ07kUvzRCXNgeAAMzFLI3xRcU1NKj4573gEPesReypsj5Kg65PlesqLOa0YOK5OYuVnFPhfZrGju3u0wEuJz5AY9H1AESl3zZ4pNn93-4ZdRJNAEqa91W88Ch_p3lYLZs9lFRVsya_fw0UnSvmp4s_LM1Ri95h9Q9s9A867cfEzJuWlmtZBEMoU_9C4RAwCCvx5vh0Nww1U6PJR7uVpa10UqrvoYofN6qOhuJa0JpzPx7zW8rN0PR_OvySpFjLWGHfezmo4Rm8mGhlFt5Jh8oWRrTkbiBshsIjSpeRVXiHqsxx8_byFIFP0wwI2wSwQ\",\"e\":\"AQAB\",\"kid\":\"j7euaytua\",\"qi\":\"SgMDCFsHdFSoy9w5X00vc9vM5p1emPoNDyJ_qbq-yS2WlfDKDmovsv4ap08Tg_T_oM1_QVyYERag4iTKVx__gCwr0R9Ty1-Ec4R0YuD5GoTh97-E-0EVNTEKvhI1q4_XnZrSXnkqRseXP8eOuNRGJadRD9tsrbfN4UCEL1S_jZM\",\"dp\":\"Rah0_svsXeBjtvJAY0OhnoA8icALvILygiS0qaYE4njsK6iP1B1Zb5oekfEq8rW4o8nReQLgCaW59HTvKrQznY53iMebG2w3dnFU4V4dAcl8XvohkZ7R0MwB3gadBlor6CcGEKJ9UaEv821aUI7h2VRYHscGkVxGveqn3CkZHwE\",\"alg\":\"RS256\",\"dq\":\"mXdq0hu1UFusF8C1H9HvHOanLxX1tBmYv5aM9d2rd5mIZhssgPSc1TlOokRnXRpAvvml6z74wCAKwekB1HAMT7oDI8rjqZc5Za6_LVmKubzP43TlSpDrBrZiQqt40KRoLA60afZajZihSthJSiZQa7NJMvDYF_YnFbJL3CgUcz0\",\"n\":\"1lXRCVqEAc2Du3Ky_qf_HK6MZ6YPinhqw_-L73ngQDN6BCwXsKRFNDreJkrCoooKvPmApsCg_4UFcOPOhkeOvjT7kp723KyHgHKtORoSuxQHYiD4ksubzWeVRJqhDSCHLbFwASnwSfzSCgw2UGUerCHwhjYF5lYXSjynTuncKIrq-d8-H2kAYjiW0NiawG_fGjeinKgwtUoVrYkRcEevikp3r8xzWmIIhuaTAwxXYx5SN6H6sXsgvHc37Fo3umMCu35J2pqGTgwQQar9UJFUm_G5W5bZHFoc_TDuA1oS7ZRmPwNIQn7FOlNz-x05yamFYfXHAyp-MYIq-98lcdf0hQ\"}" ENDPOINT="cache.cell-alpha-dev.preprod.a.momentohq.com" dotnet run --project MomentoApplicationPresignedUrl
Request claims: exp = 1649874435, cache = cache, key = MyKey, ttl (for set) = 1209600
Signed URL for SET:
https://cache.cell-alpha-dev.preprod.a.momentohq.com/cache/set/cache/MyKey?ttl_milliseconds=1209600000&token=eyJhbGciOiJSUzI1NiIsImtpZCI6Imo3ZXVheXR1YSIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NDk4NzQ0MzUsImNhY2hlIjoiY2FjaGUiLCJrZXkiOiJNeUtleSIsIm1ldGhvZCI6WyJzZXQiXSwidHRsIjoxMjA5NjAwfQ.BikiWlay95plBTuDWb2E9-77YckdNeHrjqgUCtCASLIMBDtBn9I49hNtXLPBwksJEqnKcBHRzLEz-g8cDtzpjvUgXmSlSH_dD4kiqAhgDIf9SydhjxwGbCDdV4y4P1c5yTbR8eaQnQIm1wRheAnvmsYHXNpF3sbOCUNlkYECZza76jOgodTmf_PolpK7VjpKl0FeEYez1MitNa88rBd-jKcrJbUFj36puMdhqkJI1u9mi1m-KxvAvmvZ6FSSTH3iiAa5MLpK1Fgnu4haT9cGO-Z8RwZyRi1kbbg4Zk2KthMV07Ev0cORGRC6u5RP-Wa1lRr1h7aQt37zN12d9BTMig
Signed URL for GET:
https://cache.cell-alpha-dev.preprod.a.momentohq.com/cache/get/cache/MyKey?token=eyJhbGciOiJSUzI1NiIsImtpZCI6Imo3ZXVheXR1YSIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NDk4NzQ0MzUsImNhY2hlIjoiY2FjaGUiLCJrZXkiOiJNeUtleSIsIm1ldGhvZCI6WyJnZXQiXX0.a-V515s1K7lkIh_9jK1eHdiUBEZNEsXBntlL4y-1cEgTgRwyNM8gNTztVzYfqvoP5qq2apIp_5Wg6GQeLNaEZn-vdf9Fn9MgxFSjJ0VXraHKZu34Yhg5evPEDdWV49rncybrCJ4ZFdGiJLHXOGX8usAxilQxxCVH-uZn3Dg2_OIInLIWCgR20wIDwGY4rOZ4ZBzLlhU-VuRwpyOQMlXOaTaD7TKAMJcPURXPsQ1UyoBB2YS56aQ5BAYDoJAGySEhFNmD2jfN01lyjvV6WQ_US0AhrFygHuQYCi1-d592HKJvSpuan0J2uaEm5uIBLGNfiSSSsqIUL9MZLhu4Vf0juA
Posting value with signed URL for SET: MyData
Retrieved value with signed URL for GET: MyData
```

```
% curl https://cache.cell-alpha-dev.preprod.a.momentohq.com/cache/set/cache/MyKey\?ttl_milliseconds\=1209600000\&token\=eyJhbGciOiJSUzI1NiIsImtpZCI6Imo3ZXVheXR1YSIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NDk4NzQ0MzUsImNhY2hlIjoiY2FjaGUiLCJrZXkiOiJNeUtleSIsIm1ldGhvZCI6WyJzZXQiXSwidHRsIjoxMjA5NjAwfQ.BikiWlay95plBTuDWb2E9-77YckdNeHrjqgUCtCASLIMBDtBn9I49hNtXLPBwksJEqnKcBHRzLEz-g8cDtzpjvUgXmSlSH_dD4kiqAhgDIf9SydhjxwGbCDdV4y4P1c5yTbR8eaQnQIm1wRheAnvmsYHXNpF3sbOCUNlkYECZza76jOgodTmf_PolpK7VjpKl0FeEYez1MitNa88rBd-jKcrJbUFj36puMdhqkJI1u9mi1m-KxvAvmvZ6FSSTH3iiAa5MLpK1Fgnu4haT9cGO-Z8RwZyRi1kbbg4Zk2KthMV07Ev0cORGRC6u5RP-Wa1lRr1h7aQt37zN12d9BTMig -d "somethingElse" --http1.1
{
 "result": "Ok",
 "message": ""
}

% curl https://cache.cell-alpha-dev.preprod.a.momentohq.com/cache/get/cache/MyKey\?token\=eyJhbGciOiJSUzI1NiIsImtpZCI6Imo3ZXVheXR1YSIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NDk4NzQ0MzUsImNhY2hlIjoiY2FjaGUiLCJrZXkiOiJNeUtleSIsIm1ldGhvZCI6WyJnZXQiXX0.a-V515s1K7lkIh_9jK1eHdiUBEZNEsXBntlL4y-1cEgTgRwyNM8gNTztVzYfqvoP5qq2apIp_5Wg6GQeLNaEZn-vdf9Fn9MgxFSjJ0VXraHKZu34Yhg5evPEDdWV49rncybrCJ4ZFdGiJLHXOGX8usAxilQxxCVH-uZn3Dg2_OIInLIWCgR20wIDwGY4rOZ4ZBzLlhU-VuRwpyOQMlXOaTaD7TKAMJcPURXPsQ1UyoBB2YS56aQ5BAYDoJAGySEhFNmD2jfN01lyjvV6WQ_US0AhrFygHuQYCi1-d592HKJvSpuan0J2uaEm5uIBLGNfiSSSsqIUL9MZLhu4Vf0juA
somethingElse
```

### CODE REVIEW

File: dotnet/MomentoExamples/MomentoApplicationPresignedUrl/Program.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        static async Task Main(string[] args)
+        {
+            // prepare requests
+            uint expiryEpochSeconds = (uint) DateTimeOffset.UtcNow.AddMinutes(URL_TTL_MINUTES).ToUnixTimeSeconds();
+            var setReq = new SigningRequest(CACHE_NAME, KEY, CacheOperation.SET, expiryEpochSeconds)
+            {
+                TtlSeconds = OBJECT_TTL_SECONDS
+            };
+            var getReq = new SigningRequest(CACHE_NAME, KEY, CacheOperation.GET, expiryEpochSeconds);
```
Comment (mjindalmotiv8) [inline dotnet/MomentoExamples/MomentoApplicationPresignedUrl/Program.cs:47]: So can the signingrquest be an array of operations?

Comment (wandaitzuchen) [reply dotnet/MomentoExamples/MomentoApplicationPresignedUrl/Program.cs:47]: Hmm good point, currently in our SDK (both c# and python) we only allow one single operation when creating a request. 
https://github.com/momentohq/client-sdk-csharp/blob/4ba95e8696e03e72353b160bf5f67c897c8f4040/Momento/MomentoSigner.cs#L55
https://github.com/momentohq/client-sdk-python/blob/262ac0fa655eff80f781fc135f578bf81cf487ba/src/momento/momento_signer.py#L103-L105

Not sure if there will be use cases for a token to allow both.  If so we need to allow an array of operations in SDK.

Comment (mjindalmotiv8) [reply dotnet/MomentoExamples/MomentoApplicationPresignedUrl/Program.cs:47]: I think we should - Wyze has a use case where they will generate a pre-signed url for both operations for their AI workflow.

Comment (mjindalmotiv8) [reply dotnet/MomentoExamples/MomentoApplicationPresignedUrl/Program.cs:47]: Chatted with @danielamiao  and this is a SDK side change only - we can punt it until Wyze AI team integrates with us so definitely not an immediate blocker.

**Review by mjindalmotiv8 — DISMISSED** [review]

File: dotnet/MomentoExamples/MomentoApplicationPresignedUrl/Program.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            
+            var getResponse = await client.GetStringAsync(getUrl);
+            Console.WriteLine($"Lookedup value with signed URL for {CacheOperation.GET}: {getResponse}");
+        }
+
+        static async Task Main(string[] args)
+        {
+            // prepare requests
+            uint expiryEpochSeconds = (uint) DateTimeOffset.UtcNow.AddMinutes(URL_TTL_MINUTES).ToUnixTimeSeconds();
+            var setReq = new SigningRequest(CACHE_NAME, KEY, CacheOperation.SET, expiryEpochSeconds)
```
Comment (mjindalmotiv8) [inline dotnet/MomentoExamples/MomentoApplicationPresignedUrl/Program.cs:43]: I think there are too many keys - lets name KEY to CACHE_KEY

File: dotnet/MomentoExamples/MomentoApplicationPresignedUrl/Program.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        private static string Base64Encode(string plainText) {
+            var plainTextBytes = System.Text.Encoding.UTF8.GetBytes(plainText);
+            return System.Convert.ToBase64String(plainTextBytes);
+        }
+
+        private static async Task RunPresignedUrlExample(Uri setUri, Uri getUri)
+        {
+            Console.WriteLine($"Posting value with signed URL for {CacheOperation.SET}: {OBJECT_VALUE}");
+            var json = JsonConvert.SerializeObject(Base64Encode(OBJECT_VALUE));
```
Comment (danielamiao) [inline dotnet/MomentoExamples/MomentoApplicationPresignedUrl/Program.cs:30]: Add //TODO here to highlight why base64 is needed for now, but we intend to fix it later?

**Review by danielamiao — APPROVED** [review]
Nice! I don't see an explicit set HTTP=1.1 option anywhere so I'm guessing the http library defaults to HTTP1?

**Comment by wandaitzuchen** [comment]
Yup! .NET's http client defaults to http 1.1.

**Pull request merged**